twitter_tunes/tests/test_bot.py

```diff
@@ -0,0 +1,227 @@
+                          set_redis_data):
+    """Test if main does stuff if twitter goes horribly wrong.
+
+    Make sure it can keep going."""
+    get_redis_data.return_value = {}
+    youtube_api.return_value = (u'url', True)
+    call_twitter_api.side_effect = ValueError('Missing OAuth key or token.')
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_api')
+@mock.patch('twitter_tunes.scripts.twitter_bot.youtube_api.get_link')
+@mock.patch('twitter_tunes.scripts.twitter_bot.tweepy.API.update_status')
+def test_main_bad_update(update_status, youtube_api, twitter_api,
+                         get_redis_data, set_redis_data):
+    """Test if main does stuff if tweepy goes horribly wrong.
+
+    Make sure it can keep going."""
+    from tweepy import TweepError
+    get_redis_data.side_effect = redis_side_effect
+    youtube_api.return_value = (u'url', True)
+    update_status.side_effect = TweepError("Couldn't Post")
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_api')
+@mock.patch('twitter_tunes.scripts.youtube_api')
+@mock.patch('twitter_tunes.scripts.twitter_bot.tweepy.API.trends_place')
+def test_main_bad_get_trends(trends_place, yt, tw, get, set):
+    """Test if main does stuff if tweepy goes horribly wrong.
+
+    Make sure it can keep going."""
+    from tweepy import RateLimitError
+    get.return_value = {}
+    trends_place.side_effect = RateLimitError("Slow Down!")
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_api')
+@mock.patch('tweepy.API')
+@mock.patch('twitter_tunes.scripts.twitter_bot.youtube_api.youtube_search')
+def test_main_bad_youtube(youtube_search, tweepy, ta, get, set):
+    """Test if main does stuff if twitter goes horribly wrong.
+
+    Make sure it can keep going."""
+    get.side_effect = redis_side_effect
+    from apiclient.errors import HttpError
+    youtube_search.side_effect = HttpError('Uhh', b'youtube broke.')
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+def test_main_bad_redis(get_redis_data):
+    """Test if main does stuff if redis goes horribly wrong."""
+    from requests.exceptions import ConnectionError
+    get_redis_data.side_effect = ConnectionError
+    assert twitter_bot.main() == u'Something went horribly wrong.'
```

Comment:
these tests for the bot are more instructive than that first one.  Here you've isolated one particular failure point.  That's more useful than failing a lot of things.